scripts/utils/trading_strategy_helpers.js

```diff
@@ -360,6 +361,66 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
+   * the resulting transaction is that of transfering all specified funds from master through its brackets
+   * followed by approval and deposit of those same tokens into BatchExchange on behalf of each bracket.
+   *
+   * @param {string} masterAddress Ethereum address of Master Gnosis Safe (Multi-Sig)
+   * @param {Transfer[]} transferList List of {@link Deposit} that are to be bundled together
+   * @param {boolean} [useWei=false] flag allowing user to specify if amounts are already in Base token units
+   * @param {boolean} [unsafe=false] does not perform balance verification
+   * @param {boolean} [debug=false] prints log statements when true
+   * @returns {Transaction} all the relevant transaction information used for submission to a Gnosis Safe Multi-Sig
+   */
+  const buildTransferDataFromList = async function (masterAddress, transferList, useWei = false, unsafe = false, debug = false) {
+    const log = debug ? (...a) => console.log(...a) : () => {}
+
+    const uniqueTokens = uniqueItems(transferList.map((t) => t.tokenAddress))
+    const tokenInfo = fetchTokenInfoForFlux(transferList)
+
+    // Will be used to make sufficient balance assertion before transfer.
+    const cumulativeAmounts = new Map(uniqueTokens.map((address) => [address, new BN(0)]))
+    const transactions = await Promise.all(
+      transferList.map(async (transfer) => {
+        const token = await tokenInfo[transfer.tokenAddress]
+        let unitAmount = new BN(transfer.amount)
+        if (!useWei) {
+          unitAmount = toErc20Units(transfer.amount, token.decimals)
+        }
+        // Accumulate amounts being transfered for each token.
+        cumulativeAmounts.set(token.address, cumulativeAmounts.get(token.address).add(unitAmount))
+
+        log(`Transferring ${unitAmount} ${token.symbol} to ${transfer.receiver} from ${shortenedAddress(masterAddress)}`)
+        const transferData = token.instance.contract.methods.transfer(transfer.receiver, unitAmount.toString()).encodeABI()
+        return {
+          operation: CALL,
+          to: token.address,
+          value: 0,
+          data: transferData,
+        }
+      })
+    )
+    log(`Transfer bundle contains ${transactions.length} elements and sends ${uniqueTokens.length} distinct tokens.`)
+    if (!unsafe) {
+      // Ensure sufficient funds.
+      for (const tokenAddress of uniqueTokens) {
+        const token = await tokenInfo[tokenAddress]
+        const masterBalance = await token.instance.balanceOf(masterAddress)
+        log(`Ensuring sufficient ${token.symbol} balance for this transfer...`)
+        if (masterBalance.lt(cumulativeAmounts.get(tokenAddress))) {
+          throw new Error(
+            `Master Safe has insufficient ${token.symbol} balance (${masterBalance.toString()} < ${cumulativeAmounts
+              .get(tokenAddress)
+              .toString()})`
+          )
+        }
+      }
+    }
```

Comment:
Here we await the balances of the master Safe one token after the other. You could replace the for loop with a `Promise.all`: 
```suggestion
    if (!unsafe) {
      // Ensure sufficient funds.
      await Promise.all(
        uniqueTokens.map(async (tokenAddress) => {
          const token = await tokenInfo[tokenAddress]
          const masterBalance = await token.instance.balanceOf(masterAddress)
          log(`Ensuring sufficient ${token.symbol} balance for this transfer...`)
          if (masterBalance.lt(cumulativeAmounts.get(tokenAddress))) {
            throw new Error(
              `Master Safe has insufficient ${token.symbol} balance (${masterBalance.toString()} < ${cumulativeAmounts
                .get(tokenAddress)
                .toString()})`
            )
          }
        })
      )
    }
```
For extra efficiency (~1s) you could retrieve the master balances during the "await" before this loop.